src/main/java/tr/edu/ogu/ceng/advice/ImsAOPStudent.java

```diff
@@ -27,6 +30,56 @@ public void beforeAddStudent(StudentDto studentDto) {
 		} else if (existingStudentStudenNo == true) {
 			throw new ServiceException("Bu öğrenci numarası zaten kayıtlı.");
 		}
-    }    
+	}
+
+	@Before("execution(* tr.edu.ogu.ceng.service.StudentService.registerAsStudent(..)) && args(request)")
+	public void beforeRegisterAsStudent(StudentDto request) {
+		validateStudentNo(request.getStudentNo());
+		validateTckno(request.getTckn());
+		checkIfStudentAlreadyRegistered(request.getStudentNo(), request.getTckn(), request.getEmail(),
+				request.getUsername());
+		checkIfPasswordsMatchingValidation(request);
+
+	}
```

Comment:
Bu validasyonlar addStudent için de geçerliyse eğer tek bir method altında toplayabiliriz. beforeAddStudent içinde de validateTckno, validateStudentNo gibi kontroller olması gerekli bence. Bu durumda kontrollerin hepsini beforeAddStudent methodu altında toplayıp, checkIfStudentAlreadyRegistered methodundaki kod tekrarından da kurtulabiliriz. Yoksa beforeAddStudent ve beforeRegisterAsStudent olarak ayırmak mı gerekli? @ykartal, @OmerKurtuldu Hangisi daha doğru bir çözüm olur?

Reply:
Ben, 'checkIfStudentAlreadyRegistered' fonksiyonunu başka bir yerde tanımlayarak, bu fonksiyonu iki ayrı 'before' anotasyonu ile kullanmanın daha uygun olacağını düşündüm. Ancak, arkadaşlarımın yazdığı kodu değiştirmek istemediğim için bu yöntemi uygulayamadım. @ykartal hocam nasıl bir yaklaşım daha doğru olur? Düzenlemelerimi ona göre yapabilirim.